### PR TITLE
[FX-5661] Migrate UserBadge

### PR DESCRIPTION
**Pull request by sashuk**
[FX-5661]

### Description

This pull request migrates UserBadge to TailwindCSS.

### How to test

- [Temploy](https://picasso.toptal.net/fx-5661-userbadge)
- Compare temploy with https://picasso.toptal.net/?path=/story/components-userbadge--userbadge
- [Visual diffs](https://happo.io/a/675/p/848/compare/54723b5b9999e497b9b734ae99c24814890e65fd/f060d08e763900a65ed2588778635d13e9080305) are only for case when image is not loaded, so does not affect regular component

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5661]: https://toptal-core.atlassian.net/browse/FX-5661?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

File: .changeset/tasty-avocados-shop.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@toptal/picasso-user-badge': major
```
Comment (dmaklygin) [inline .changeset/tasty-avocados-shop.md:2]: shouldn't we update picasso as well?

**Review by dmaklygin — APPROVED** [review]

File: packages/base/UserBadge/src/UserBadge/UserBadge.tsx [hunk 2]
```diff
@@ -112,11 +101,11 @@ export const UserBadge = forwardRef<HTMLDivElement, Props>(function UserBadge(
       ref={ref}
       flex
       alignItems={alignItems}
-      className={cx(classes.root, className)}
+      className={twMerge('text-lg', className)}
       style={style}
```
Comment (AdrianContiu) [inline packages/base/UserBadge/src/UserBadge/UserBadge.tsx:105]: `text-lg` also sets the `line-height` to 1.5rem, which was not set before. Perhaps this is also the reason for the failed Happo tests in Cypress.

**Review by AdrianContiu — APPROVED** [review]
I left a comment that hopefully resolves the differences in the Happo tests.

File: .changeset/tasty-avocados-shop.md (hunk 1 above)
Comment (sashuk) [reply .changeset/tasty-avocados-shop.md:2]: @dmaklygin Can you please comment on https://github.com/toptal/picasso/pull/4497#discussion_r1726991346? I think we should not, but let's double check

File: packages/base/UserBadge/src/UserBadge/UserBadge.tsx (hunk 2 above)
Comment (sashuk) [reply packages/base/UserBadge/src/UserBadge/UserBadge.tsx:105]: @AdrianContiu Thank you, the problem was addressed!

**Comment by sashuk** [comment]
Merging, but still curious to discuss the https://github.com/toptal/picasso/pull/4501#discussion_r1728642662

**Pull request merged**